src/NServiceBus.Core/Transports/Learning/LearningTransportInfrastructure.cs

```diff
@@ -43,23 +42,31 @@ public LearningTransportInfrastructure(SettingsHolder settings)
 
         public override OutboundRoutingPolicy OutboundRoutingPolicy { get; } = new OutboundRoutingPolicy(OutboundRoutingType.Unicast, OutboundRoutingType.Multicast, OutboundRoutingType.Unicast);
 
-        string FindSolutionRoot()
+        static string FindStoragePath()
         {
             var directory = AppDomain.CurrentDomain.BaseDirectory;
 
             while (true)
             {
+                // Finding a solution file takes precedence
                 if (Directory.EnumerateFiles(directory).Any(file => file.EndsWith(".sln")))
                 {
-                    return directory;
+                    return Path.Combine(directory, DefaultLearningTransportDirectory);
+                }
+
+                // When no solution file was found try to find a learning transport directory
+                // don't ignore casing due to linux support
+                if (Directory.EnumerateDirectories(directory).Any(dir => dir.Equals(DefaultLearningTransportDirectory, StringComparison.Ordinal)))
+                {
+                    return Path.Combine(directory, DefaultLearningTransportDirectory);
                 }
 
                 var parent = Directory.GetParent(directory);
 
                 if (parent == null)
                 {
                     // throw for now. if we discover there is an edge then we can fix it in a patch.
-                    throw new Exception("Couldn't find the solution directory for the learning transport. If the endpoint is outside the solution folder structure, make sure to specify a storage directory using the 'EndpointConfiguration.UseTransport<LearningTransport>().StorageDirectory()' API.");
+                    throw new Exception($"Couldn't find the solution directory or a directory named '{DefaultLearningTransportDirectory}' for the learning transport. Make sure either a '{DefaultLearningTransportDirectory}' exists in the current or one of the parent directories or specify a storage directory using the 'EndpointConfiguration.UseTransport<LearningTransport>().StorageDirectory()' API.");
```

Comment:
Suggested tweak to the exception message with some assistance from David:

```suggestion
                    throw new Exception($"Unable to determine the storage directory path for the learning transport due to the absence of a solution file. Either create a '{DefaultLearningTransportDirectory}' directory in one of this project’s parent directories, or specify the path explicitly using the 'EndpointConfiguration.UseTransport<LearningTransport>().StorageDirectory()' API.");
```